raiden/network/proxies/token_network.py

```diff
@@ -623,6 +634,13 @@ def _deposit_preconditions(
         if not isinstance(total_deposit, int):
             raise ValueError('total_deposit needs to be an integral number.')
 
+        if not self.client.can_query_state_for_block(block_identifier):
+            # This can happen if we shutdown with a deposit transaction saved
+            # in our pending transactions and restart after 128 blocks. Then
+            # we have no choice but to query the latest state to calculate the
+            # deposit amounts
+            block_identifier = 'latest'
```

Comment:
latest **is not a valid value**, that is *the reason why we have done the work for the consistent blockchain view.* (This also applies to the other cases in this PR)

Reply:
additionally, why the handling here is different from the other places? specifically some functions raise `InvalidBlockNumberInput` while this changes the block number. (This also applies to the cases that just skip the check altogether)

Reply:
Handling will be made consistent.